webpack/production.js

```diff
@@ -0,0 +1,61 @@
+const path = require('path');
+const webpack = require('webpack');
+
+/**
+ * These are all webpack plugins used to make work
+ * long term caching
+ */
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const ManifestPlugin = require('webpack-manifest-plugin');
+const ChunkManifestPlugin = require('chunk-manifest-webpack-plugin');
+const WebpackMd5Hash = require('webpack-md5-hash');
+const InlineManifestWebpackPlugin = require('inline-manifest-webpack-plugin');
+
+const resolvePath = require('./helpers/resolve-path');
+
+module.exports = {
+  entry: {
+    app: resolvePath('src/index.js')
+  },
+  output: {
+    path: path.join(__dirname, 'build'),
+    filename: '[name].[chunkhash].js',
+    chunkFilename: '[name].[chunkhash].js'
+  },
+  resolve: {
+    extensions: [ '.js', '.jsx' ],
+    modules: [
+      resolvePath('node_modules'),
+      resolvePath('src')
+    ]
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        query: { cacheDirectory: true }
+      },
+    ],
+  },
+  plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+      name: 'vendor',
+      chunks: ['app'],
+      minChunks: ({ resource }) => /node_modules/.test(resource)
```

Comment:
What I'm trying to do here is [dynamically move all node_modules](https://webpack.js.org/plugins/commons-chunk-plugin/#passing-the-minchunks-property-a-function) dependencies into `vendor` chunk.

I'm not sure if it's a good idea because my goal is to implement code splitting. That means that maybe some dependency is just used in one of our dynamic chunks (for example on `route-b`. Maybe it's a bad idea to have that underused dependency in vendor. We'll see.